hazelcast/src/main/java/com/hazelcast/map/impl/querycache/accumulator/DefaultCyclicBuffer.java

```diff
@@ -137,16 +138,11 @@ public int size() {
         long head = headSequence.get();
         long tail = tailSequence.get();
 
-        if (tail == UNAVAILABLE) {
+        if (tail == UNSET) {
             return 0;
         }
 
-        int avail = (int) (tail - head + 1);
-        if (avail <= capacity) {
-            return avail;
-        } else {
-            return capacity;
```

Comment:
How was it even possible in the previous code that the available was higher than the capacity?